### ISSUE TITLE
[Bug]: Runtime errors: SDK state invalid -- undefined provider

### ISSUE THREAD

**Issue by NoahJinnn** [MetaMask/metamask-sdk#671]
### Provide environment information

- Mac M1
- Node v18.19.0
- Yarn 1.22.21

### MetaMask SDK Version

@metamask/sdk-react: 0.14.2

### MetaMask Mobile app Version

not relevant

### What browser are you using? (if relevant)

Chrome Version 121.0.6167.85

### How are you deploying your application? (if relevant)

Nginx

### Describe the Bug

I wanted to detect if the users installed MetaMask or not, but the bug crashed my app completely. This bug happens on my server which is using Nginx and `localhost`. Information on how it runs on the server:

<img width="861" alt="Screenshot 2024-02-07 at 11 12 46" src="https://github.com/MetaMask/metamask-sdk/assets/41507250/36f1a1a9-9c0d-4576-b295-20153d0b9087">

### Expected Behavior

Should show the dialog to remind MetaMask installation.

<img width="667" alt="Screenshot 2024-02-07 at 11 10 41" src="https://github.com/MetaMask/metamask-sdk/assets/41507250/2a70755a-46fa-44a0-a21a-f902d6552501">

### To Reproduce

- Make sure your browser doesn't install MetaMask wallet
- Install Phantom wallet https://chromewebstore.google.com/detail/phantom/bfnaelmomeimhlpmgjnjophhpkkoljpa
- Create a wallet
- Clone https://github.com/NoahJinnn/react-metamask-sdk
- Run `yarn start`

<img width="1799" alt="Screenshot 2024-02-07 at 11 03 15" src="https://github.com/MetaMask/metamask-sdk/assets/41507250/dcb1fd6e-35b6-4cd0-9df9-905180674b39">

### However, the bug won't happen if we serve the build:

```sh
yarn build
serve -s build # Use the Network link, if we use Local link the bug will appear
```

<img width="452" alt="Screenshot 2024-02-07 at 11 03 55" src="https://github.com/MetaMask/metamask-sdk/assets/41507250/4974da06-dc02-4973-a3e2-b1e1751674b3">

<img width="1575" alt="Screenshot 2024-02-07 at 11 04 21" src="https://github.com/MetaMask/metamask-sdk/assets/41507250/092fb922-528a-4f3c-af93-b4e3736cd23b">

**Comment by minhDaiVatek** [comment]
+1

**Comment by christopherferreira9** [comment]
Hi all! 
We are aware of this warning message.
This is only shown in development mode, if you  do `sdk?.terminate()` and refresh you'll clean up this message

**Comment by andreahaku** [comment]
@NoahJinnn closing as the issues should be fixed with 0.17.1.
Thanks
